pkg/activator/handler/handler.go

```diff
@@ -54,5 +54,28 @@ func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	proxy := httputil.NewSingleHostReverseProxy(target)
 	proxy.Transport = a.Transport
 
+	setupPruningHeaders(proxy)
+
 	proxy.ServeHTTP(w, r)
 }
+
+func setupPruningHeaders(p *httputil.ReverseProxy) {
+	headersToRemove := []string{
+		activator.ConfigurationHeader,
+		activator.RevisionHeaderName,
+		activator.RevisionHeaderNamespace,
+	}
+
+	orig := p.Director
+	p.Director = func(r *http.Request) {
```

Comment:
For my own understanding: Is it not possible to just drop the headers from the incoming request? Is it done this way to keep the incoming request object intact so other methods can happily use the headers just like today?